homeassistant/components/hifiberry/media_player.py

```diff
@@ -0,0 +1,262 @@
+"""
+HifiBerry Platform.
+
+HifiBerry rest API: https://github.com/hifiberry/audiocontrol2/blob/fee165140c9da044c2166cac1d77ae5e0008c351/doc/api.md
+"""
+
+from datetime import timedelta
+import logging
+import socket
+import requests
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import PLATFORM_SCHEMA, MediaPlayerDevice
+from homeassistant.components.media_player.const import (
+    MEDIA_TYPE_MUSIC,
+    SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE,
+    SUPPORT_PLAY,
+    SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_STOP,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP,
+)
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PORT,
+    STATE_IDLE,
+    STATE_PAUSED,
+    STATE_PLAYING,
+)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "HifiBerry"
+DEFAULT_PORT = 81
+
+DATA_HIFIBERRY = "hifiberry"
+
+TIMEOUT = 10
+SCAN_INTERVAL = timedelta(seconds=1)
+
+SUPPORT_HIFIBERRY = (
+    SUPPORT_PAUSE
+    | SUPPORT_VOLUME_SET
+    | SUPPORT_VOLUME_MUTE
+    | SUPPORT_PREVIOUS_TRACK
+    | SUPPORT_NEXT_TRACK
+    | SUPPORT_STOP
+    | SUPPORT_PLAY
+    | SUPPORT_VOLUME_STEP
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler